### PR TITLE
Updated RecoveryServices.Backup from latest swagger spec

### PR DESCRIPTION
**Pull request by DheerendraRathor**
## Description

- Moved ListJobs and GetJobDetails to api version 2017-07-01
- Bumped version to 2.1.0-preview

Spec PR: https://github.com/Azure/azure-rest-api-specs/pull/1502

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@DheerendraRathor add a description as to what this PR is all about.
Any information that helps in understanding what is this PR about.
If it's just API version change, mention that in the description.

**Comment by DheerendraRathor** [comment]
@shahabhijeet I've updated description. Looks like I don't have permission to update labels.

**Comment by DheerendraRathor** [comment]
@shahabhijeet Any update on this?

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Recovery Services Backup Resources.")]
 
 [assembly: AssemblyVersion("1.0.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs]: @DheerendraRathor bump Assemblyversion to 2.0.0.0

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
When you now execute generate.cmd, it creates a .txt file under _metadata directory for your RP, which will be helpful for all future regeneration of the SDK

**Comment by DheerendraRathor** [comment]
@shahabhijeet Added _metadata/recoveryservicesbackup_resource-manager.txt and updated AssemblyVersion

**Comment by DheerendraRathor** [comment]
@shahabhijeet Any update on this PR?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**